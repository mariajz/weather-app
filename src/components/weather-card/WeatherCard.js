import React from 'react';
import { WeatherCardWrapper, StyledCustomImage } from './WeatherCard.style';
import { degreeSymbol } from '../../constants/Constants';
import { StyledText } from '../../commons/styles';

const WeatherCard = ({ weatherItem }) => {
    const { condition, temp, time, weatherImage } = weatherItem;

    return (
        <>
            <WeatherCardWrapper>
                <StyledText size={14}>{time}</StyledText>
                <StyledCustomImage path={weatherImage} width={40} height={40} />
                <StyledText size={14}>{condition}</StyledText>
                <StyledText size={12}>
                    {temp}
                    {degreeSymbol}C
                </StyledText>
            </WeatherCardWrapper>
        </>
    );
};

export default WeatherCard;
