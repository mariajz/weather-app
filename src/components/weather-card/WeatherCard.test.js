import { render } from '@testing-library/react-native';
import React from 'react';
import WeatherCard from './WeatherCard';

const defaultProps = {
    weatherItem: {
        condition: 'Clear',
        temp: 24.9,
        time: '12 AM',
        weatherImage: 10,
    },
};
describe('WeatherCard', () => {
    it('should render WeatherCard', () => {
        const container = render(<WeatherCard {...defaultProps} />);

        expect(container).toMatchSnapshot();
    });
});
