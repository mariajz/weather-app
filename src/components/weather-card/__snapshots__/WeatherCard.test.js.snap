// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherCard should render WeatherCard 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "rgba(255,255,255, 0.2)",
      "borderBottomLeftRadius": 5,
      "borderBottomRightRadius": 5,
      "borderTopLeftRadius": 5,
      "borderTopRightRadius": 5,
      "height": 150,
      "justifyContent": "space-between",
      "marginRight": 10,
      "paddingBottom": 10,
      "paddingLeft": 10,
      "paddingRight": 10,
      "paddingTop": 10,
      "width": 100,
    }
  }
>
  <Text
    size={14}
    style={
      {
        "color": "white",
        "fontSize": 14,
        "marginLeft": 0,
      }
    }
  >
    12 AM
  </Text>
  <Image
    height={40}
    source={10}
    style={
      {
        "height": 40,
        "width": 40,
      }
    }
    testID="image"
    width={40}
  />
  <Text
    size={14}
    style={
      {
        "color": "white",
        "fontSize": 14,
        "marginLeft": 0,
      }
    }
  >
    Clear
  </Text>
  <Text
    size={12}
    style={
      {
        "color": "white",
        "fontSize": 12,
        "marginLeft": 0,
      }
    }
  >
    24.9
    Â°
    C
  </Text>
</View>
`;
